### PR TITLE
devices: Update db devices when creating an image (update) or when upgrading a device.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
We need to update devices update availability when an image created (for all the devices that use that image set). 
and When a device is upgraded we need to update the device image and update availability.

Fixes # THEEDGE-1913

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/updates.go [hunk 1]
```diff
@@ -472,3 +476,60 @@ func (s *UpdateService) SendDeviceNotification(i *models.UpdateTransaction) (Ima
 	}
 	return notify, nil
 }
+
+// UpdateDevicesFromUpdateTransaction update device with new image and update availability
+func (s *UpdateService) UpdateDevicesFromUpdateTransaction(update models.UpdateTransaction) error {
```
Comment (AaronH88) [inline pkg/services/updates.go:481]: Overall looks great!
This function name is a little confusing, do you have any alternatives?

Perhaps `SetUpdateFlagOnDevices` , or something else :smile:

Comment (ldjebran) [reply pkg/services/updates.go:481]: we are updating only update availability , but also images, any way this must a general function to update a device from update transaction , really nothing  coming in mind than the direct function goal, what is "update the devices from update transaction" :)

Any other suggestion ?

Comment (AaronH88) [reply pkg/services/updates.go:481]: Yeah, I dont have any clearer ideas :smile: 

This name will be OK

File: pkg/services/errors.go [hunk 2]
```diff
@@ -119,3 +119,10 @@ type DeviceHasImageUndefined struct{}
 func (e *DeviceHasImageUndefined) Error() string {
 	return "device has image undefined"
 }
+
+// UpdateTransactionHasCommitUndefined indicate that the transaction has no commit defined
+type UpdateTransactionHasCommitUndefined struct{}
+
+func (e *UpdateTransactionHasCommitUndefined) Error() string {
+	return "update transaction has not no defined commit"
+}
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/errors.go:128]: @Avielyo10 that error is defined the same way all the module errors in that file,

File: pkg/services/updates.go (hunk 1 above)
Comment (ldjebran) [reply pkg/services/updates.go:481]: @Avielyo10 please propose a name :)

File: pkg/services/errors.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/errors.go]: make changes as per your recommendation.

**Pull request merged**